python/setup.py

```diff
@@ -54,23 +55,28 @@ def get_llvm():
         ftpstream = urllib.request.urlopen(url)
         file = tarfile.open(fileobj=ftpstream, mode="r|xz")
         file.extractall(path=dir)
-    return llvm_include_dir, llvm_library_dir
+    return llvm_dir
 
 
 class CMakeExtension(Extension):
-    def __init__(self, name, path, sourcedir=""):
+    def __init__(self, name, path, externlib_dir, sourcedir=""):
         Extension.__init__(self, name, sources=[])
         self.sourcedir = os.path.abspath(sourcedir)
         self.path = path
+        self.externlib_dir = os.path.abspath(externlib_dir)
 
 
 class CMakeBuild(build_ext):
 
-    user_options = build_ext.user_options + [('base-dir=', None, 'base directory of Triton')]
+    user_options = build_ext.user_options + [('base-dir=', None, 'base directory of Triton'),
+                                             ('cuda-dir=', None, 'CUDA directory'),
+                                             ('enable-libdevice', None, 'enable libdevice')]
 
     def initialize_options(self):
         build_ext.initialize_options(self)
         self.base_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))
+        self.cuda_dir = "/usr/local/cuda"
```

Comment:
Alright, that actually simplifies this work by removing the need of building `llvm-dis` (and `libncurses5`).

Reply:
Now we can point out libdevice dynamically.
Providing cuda_dir here is probably fine as we just want to allow users to specify their custom CUDA home, which could be `/scratch/user/cuda`, `/opt/cuda`, or other directories. It does not conflict with PTX path since users can still use `TRITON_PTXAS_PATH` to specify the location of `ptxas`.

Reply:
But if libdevice is specified dynamically, then it sounds like we should be re-using `TRITON_PTXAS_PATH` to get the cuda path, which would make specifying `cuda_dir` here obsolete. We can also provide an additional environment variable `TRITON_LIBDEVICE_PATH`. This is probably better than specifying cuda-path at install time, since most people install the package from pip.

Reply:
I was just afraid that install could fail if CUDA is not available on the default path.

`cuda_dir` can be specified even when install the package from pip, but it is kind tricky.

Reply:
Ah I see now. Triton installation doesn't actually depend on system CUDA headers anymore, so `['/usr/local/cuda/include']` should be deleted rather than replaced by something more portable.

Reply:
Got you